src/states/common/approved.rs

```diff
@@ -296,7 +306,18 @@ pub trait Approved: Base {
                         self, event
                     );
                 }
-                AccumulatingEvent::Relocate(payload) => self.handle_relocate_event(payload)?,
+                AccumulatingEvent::Relocate(payload) => {
+                    if let Some(signature) = signature {
```

Comment:
Would it make sense to have a subfunction here to do that check & log_or_panic since that function is big?